sdk/aqueduct/constants/enums.py

```diff
@@ -169,6 +169,10 @@ class SerializationType(str, Enum, metaclass=MetaEnum):
     BSON_TABLE = "bson_table"
     JSON = "json"
     PICKLE = "pickle"
+    # Same as pickle. Indicates that, if the object is a collection that requires
+    # additional serialization one level down, then any tables inside are serialized
+    # as BSON_TABLE.
```

Comment:
Because any tables serialized when `derived_from_bson=True` must use `BSON_TABLE` serialization. BSON_PICKLE is sort of a combination of `PICKLE` + `derived_from_bson=True`, so it would immediately imply that any tables inside will inherit the `derived_from_bson` flag.

I agree this is decently confusing. I can look into adding `derived_from_bson` as an explicit argument into `deserialize()` and serialize()` - I didn't like this before for some reason but I think I see a path through now.

Reply:
That'd be great thx!